general/array.hpp

```diff
@@ -235,18 +217,18 @@ class Array : public BaseArray
    T Min() const;
 
    /// Sorts the array. This requires operator< to be defined for T.
-   void Sort() { std::sort((T*) data, (T*) data + size); }
+   void Sort() { std::sort((T*)data, data + size); }
```

Comment:
With `MFEM_DEBUG=YES` you will get an error if the host pointer is not **the only** valid pointer -- there is an `MFEM_ASSERT` in the conversion functions from `Memory<T>` to `T*`.

Without `MFEM_DEBUG=YES`, it will simply sort the current contents of the host pointer.

Generally, we can either:
1) say that this method can be used only if one is certain that the `Array` was never moved to device, or after calling `ReadWriteAccess(false)` (which will bring the data to the host and mark the device pointer as invalid) or
2) we can add the call inside `Sort()` and let it bring the data to host automatically.

Probably the latter is the better choice (if we ever need to sort arrays that are moved to device before that), at least until we decide we need to implement `Sort()` on the device.